com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurvesElement.cs

```diff
@@ -0,0 +1,98 @@
+using UnityEngine.Formats.Alembic.Sdk;
+
+namespace UnityEngine.Formats.Alembic.Importer
+{
+    class AlembicCurvesElement : AlembicElement
+    {
+        // members
+        aiCurves m_abcSchema;
+        PinnedList<aiCurvesData> m_abcData = new PinnedList<aiCurvesData>(1);
+        aiCurvesSummary m_summary;
+        aiCurvesSampleSummary m_sampleSummary;
+
+        internal override aiSchema abcSchema { get { return m_abcSchema; } }
+        public override bool visibility
+        {
+            get { return m_abcData[0].visibility; }
+        }
+
+        internal override void AbcSetup(aiObject abcObj, aiSchema abcSchema)
+        {
+            base.AbcSetup(abcObj, abcSchema);
+            m_abcSchema = (aiCurves)abcSchema;
+            m_abcSchema.GetSummary(ref m_summary);
+        }
+
+        public override void AbcPrepareSample()
+        {
+            base.AbcPrepareSample();
+        }
+
+        public override void AbcSyncDataBegin()
+        {
+            if (!m_abcSchema.schema.isDataUpdated)
+                return;
+
+            var sample = m_abcSchema.sample;
+            sample.GetSummary(ref m_sampleSummary);
+
+            // get points cloud component
+            var curves = abcTreeNode.gameObject.GetComponent<AlembicCurves>();
+            if (curves == null)
+            {
+                curves = abcTreeNode.gameObject.AddComponent<AlembicCurves>();
+                //  abcTreeNode.gameObject.AddComponent<AlembicPointsRenderer>(); // Need rendering
```

Comment:
You should probably remove the commented code and squash the 4 lines into a single GetOrAddComponent